plugin/semisync/semisync_master_plugin.cc

```diff
@@ -165,6 +166,11 @@ static int repl_semi_before_send_event(Binlog_transmit_param *param,
                                        const char *log_file, my_off_t log_pos) {
   if (!is_semi_sync_dump()) return 0;
 
+  if (param->server_id > 0 &&
+      !repl_semisync->verify_against_whitelist(param->server_id)) {
```

Comment:
This check wasn't in the original patch. Is it necessary to check against the whitelist prior to sending? It seems we check on both send and receive then for the common use case where the semi-sync slave is in the whitelist.

Reply:
This check is not necessary, but I wanted to stop binlog sender before calling `updateSyncHeader()`  in `repl_semi_before_send_event()`.

Reply:
It's removed now.